product_attribute_global_item_code/product.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+#
+#   Copyright (C) 2015 Akretion (http://www.akretion.com). All Rights Reserved
+#   @author Abdessamad HILALI <abdessamad.hilali@akretion.com>
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as
+#   published by the Free Software Foundation, either version 3 of the
+#   License, or (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp import fields, models, api, _
+from openerp.exceptions import Warning
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    base_code = fields.Char(
+        'Base Code',
+        help="this field is used like a base to automatically create "
+             "Internal Reference (default_code)")
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    use_internal_ref = fields.Boolean('Manual Internal Reference',
```

Comment:
consider renaming to use_manual_ref or use_manual_internal_ref . It is a bit confusing otherwise.